### PR TITLE
garden: new notifications

### PR DESCRIPTION
**Pull request by jamesacklin**
- Refactors the Notification component significantly and adds more display logic
- Adds several components and state logics we used in tloncorp/landscape-apps
- Brings the ShipName and Avatar components up to date, using data from contact-store and respecting calmengine settings

fixes tloncorp/landscape-apps#1689

### CODE REVIEW

File: ui/src/nav/notifications/Notification.tsx [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
 }
 
-function getNotificationType(rope: Rope) {
+function getNotificationType(rope: Rope): NotificationType {
+  if (
+    rope.thread.endsWith('/channel/edit') ||
+    rope.thread.endsWith('/channel/add') ||
+    rope.thread.endsWith('/channel/del') ||
+    rope.thread.endsWith('/joins') ||
+    rope.thread.endsWith('/leaves')
```
Comment (tomholford) [inline ui/src/nav/notifications/Notification.tsx:35]:
```suggestion
    [
      '/channel/edit',
      '/channel/add',
      '/channel/del',
      '/joins',
      '/leaves'
    ].some((thread) => rope.thread.endsWith(thread))
```

**Review by arthyn — APPROVED** [review]
I think it's ok to ship as is, but the more I think about this the more I think we should just add a type field that's just an open string like you brought up when building this. There's a lot of manual stuff here that is just too brittle and should be avoided. Content is the most egregious and I'd be curious what ideas we can come up with there to make it easier to work with.

Also we really shouldn't use `any` if we can avoid, especially in props.

**Review by arthyn — APPROVED** [review]
lgtm

**Review by patosullivan — APPROVED** [review]
Nice

**Pull request merged**